src/text/Heading.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react"
+
+export interface HeadingProps {
+  /** Heading text */
+  children: React.ReactNode
+  /** Typography style of the heading */
+  typography?: "heading-1" | "heading-2" | "heading-3" | "heading-4" | "subheading" | "strong" | "label" | "small"
+  /** Heading level (`h1` through `h4`) */
+  priority?: 1 | 2 | 3 |4
```

Comment:
I don't think we should restrict this for a11y. Consumers might have lots of content they put together with various atoms that could lead to an h5/h6 level.